src/goto-harness/recursive_initialization.cpp

```diff
@@ -1008,6 +1008,21 @@ code_blockt recursive_initializationt::build_function_pointer_constructor(
   for(const auto &target : targets)
   {
     auto const assign = code_assignt{dereference_exprt{result}, target};
+    auto sym_to_lookup =
+      target.id() == ID_address_of
+        ?
+        // This is either address of or pointer; in pointer case, we don't
+        // need to do anything
+        to_symbol_expr(to_address_of_expr(target).operands()[0])
```

Comment:
:no_entry_sign: Having converted to `address_of_exprt`, you should use `.object()`
```suggestion
        to_symbol_expr(to_address_of_expr(target).object())
```